src/lib/markbind/src/patches/htmlparser2.js

```diff
@@ -261,6 +261,22 @@ Tokenizer.prototype._stateBeforeSpecial = function(c) {
 	this._special = result + 1;
 };
 
+/**
+ * Removes the special state if the special tag was self-closed.
+ */
+Tokenizer.prototype._stateInSelfClosingTag = function(c) {
+	if (c === ">") {
+		this._cbs.onselfclosingtag();
+		this._state = TEXT;
+		this._sectionStart = this._index + 1;
+		// Allow all special tags (including script & style tags) to be self-closed
```

Comment:
Let's explain why we allow this despite the spec not allowing it. I think just stating the fact that the original htmlparser allows it is sufficient.

Reply:
To clarify, did you mean the `original htmlparser disallows it` ( that is the case, we're breaking away from the parser's default behaviour here )?

Reply:
Oops, I actually thought that the original htmlparser allowed it, since the changes we made didn't seem to affect any of the logic specific to script/style tags. After taking a closer look at the rest of the changes in our patches, it seems like you are right 🙂